lib/flop/builder.ex

```diff
@@ -142,6 +151,63 @@ defmodule Flop.Builder do
     end
   end
 
+  # credo:disable-for-next-line
+  defp runtime_dynamic(c, %module{}, field, %Filter{op: op, value: value})
+       when op in [:empty, :not_empty] do
+    field_type = module.__schema__(:type, field)
+
+    case {field_type, op, value} do
+      {{:array, _}, :empty, true} ->
+        dynamic(
+          [r],
+          ^c and
+            fragment(
+              "? = '{}' OR ? IS NULL",
+              field(r, ^field),
+              field(r, ^field)
+            )
+        )
```

Comment:
I would prefer not to use fragments, since the query strings are specific to the database is used and may not work for all databases that Ecto supports. 

Ecto complains when you use `[]` in a query because it doesn't know how to cast it, but this seems to work:

```elixir
{{:array, _} = ecto_type, :empty, true} ->
        dynamic(
          [r],
          ^c and
            (is_nil(field(r, ^field)) or
               field(r, ^field) == type(^[], ^ecto_type))
        )
```